src/dsp_tools/utils/xmlupload/upload_config.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import dataclasses
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+
+import regex
+
+from dsp_tools.utils.create_logger import get_logger
+
+logger = get_logger(__name__)
+
+
+def _transform_server_url_to_foldername(server: str) -> str:
+    """
+    Take the server URL and transform it so that it can be used as foldername.
+
+    Args:
+        server: server, e.g. "https://api.test.dasch.swiss/" or "http://0.0.0.0:3333"
+
+    Returns:
+        simplified version, e.g. "test.dasch.swiss" or "localhost"
+    """
+    server_substitutions = {
+        r"https?://": "",
+        r"^api\.": "",
+        r":\d{2,5}/?$": "",
+        r"/$": "",
+        r"0.0.0.0": "localhost",
+    }
+    for pattern, repl in server_substitutions.items():
+        server = regex.sub(pattern, repl, server)
+    return server
+
+
+@dataclass(frozen=True)
+class UploadConfig:
+    """Configuration for the upload process."""
+
+    verbose: bool = False
+    dump: bool = False
+    save_metrics: bool = False
+    preprocessing_done: bool = False
+    server_as_foldername: str = field(default="unknown")
+    save_location: Path = field(default_factory=lambda: Path.home() / Path(".dsp-tools") / "xmluploads")
+    timestamp_str: str = field(default_factory=lambda: datetime.now().strftime("%Y-%m-%d_%H%M%S"))
```

Comment:
Do you really need a factory for these two, or would a normal `default` value work, too?

Reply:
I think you're right. for the date, the semantics of it will change ever so slightly, as it will capture the date and time of loading this class to memory, rather than the date and time of instantiating the class. But that doesn't matter here, right? It's just about having the same timestamp everywhere

Reply:
I wasn't aware of this subtle (but important) difference between `default` and `default_factory`. But you're right, in our case that doesn't matter.

Reply:
the may reason for default-factory is e.g. with lists: if you use `default=[]` then all instances of the dataclass will share the same list, which is practically never what you want; on the other hand, `default_factory=list` will be called once for each instance, and hence each instance gets its own list